app/models/order.rb

```diff
@@ -13,6 +14,18 @@ class Order < ActiveRecord::Base
   validates :price, :quantity, numericality: { greater_than: 0 }
   validates :type, inclusion: { in: %w(BuyOrder SellOrder) }
 
+  def cancel!
```

Comment:
I think "if there is a bang method there must exist a non-bang method that's safer" is probs the best way to think about it.

However if I had used `update!` within `cancel!` then I think that would also be a very good use.